{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Goal: examine all 1kP algal sequences to see if these contain only an M Domain or also IKC domains."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "inseq=MIKC_orthogroup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "grep '^>' data/$inseq.fasta  | grep -F -f ./data/1kP_algae_IDs.txt  | sort | uniq > ./data/\"$inseq\"_selection-algae.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "125 ./data/MIKC_orthogroup_selection-algae.txt\n",
      ">ACRY-2040985-Pteromonas_sp.\n",
      ">AJAU-2006740-Helicodictyon_planctonicum\n",
      ">AJUW-2052652-Chloromonas_rosa\n",
      ">AKCR-2016810-Parachlorella_kessleri\n",
      ">ALZF-2019222-Halochlorococcum_marinum\n",
      ">AZZW-2004234-Chlorokybus_atmophyticus\n",
      ">BAJW-2009533-Isochrysis_sp.\n",
      ">BAJW-2010911-Isochrysis_sp.\n",
      ">BAJW-2033085-Isochrysis_sp.\n",
      ">BAJW-2043867-Isochrysis_sp.\n"
     ]
    }
   ],
   "source": [
    "wc -l ./data/\"$inseq\"_selection-algae.txt\n",
    "head ./data/\"$inseq\"_selection-algae.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "grep --no-group-separator -A 1 -F -f ./data/\"$inseq\"_selection-algae.txt data/\"$inseq\"_linear.fasta > data/\"$inseq\"_selection-algae.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">QXSZ-2000951-Mantoniella_squamata\n",
      "NAQVTFTKRKNGLMKKAMELSVLCDCQIALVIFNSNNKLFQYSSGDIKAVLERFKNDTLGPHEKRNNKDL\n",
      ">IRZA-2060948-Proteomonas_sulcata\n",
      "MGRKKIKIARINDERSRHATFAKRKNGLVKKAIELSILCDCEIALVIFNSQGKLTQYASGNIDQTLSKF\n",
      ">MXDS-2024758-Spermatozopsis_exsultans\n",
      "RVKIERIADEKLRQATFARRKNGLMRRAMELSVLCDCNVALIIFDASGRLAQYSSADMEALLEQYSRACAEPHERRTNDDLFKHQF\n",
      ">VBLH-2005464-Cladophora_glomerata\n",
      "MGRNKIKIQRIKDTRSRVTFTKRKNGLYKKLMELSVLCNAIGLVLFNENGKLFQYCSNGDMEDLLRKYNECNGKIHENR\n",
      ">AZZW-2004234-Chlorokybus_atmophyticus\n",
      "MGRAKIEIRRIEHATSRQVTFSKRRSGLLKKAYELSVLCDADIAVIMFSPTGKLFEYANSSIEDILERYRNTPKEYRDKRKFDNTDYLAKESKKLRHDLEQAKRELKYMHGEGLDGLSMQELEKVESKLEEGLRKLRTCKSNLMQQEIDRLRRQ\n"
     ]
    }
   ],
   "source": [
    "head data/\"$inseq\"_selection-algae.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ok so that's all my algal sequences from the 1kP MADS orhtogroup. Now let's add some guide sequences in which I can recognize the domains."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "76\n"
     ]
    }
   ],
   "source": [
    "grep -c '>' data/Zhang-2019-fig4_MIKC_Azfi-gymnosperms_selection-v3.faa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat data/outgroup_algae.fasta data/Zhang-2019-fig4_MIKC_Azfi-gymnosperms_selection-v3.faa data/\"$inseq\"_selection-algae.fasta > data/\"$inseq\"_selection-algae_guidev3.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(phylogenetics) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda activate phylogenetics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.458 (2020/Mar/27)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "(phylogenetics) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ginsi --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "rescale = 1\n",
      "All-to-all alignment.\n",
      "tbfast-pair (aa) Version 7.458\n",
      "alg=L, model=BLOSUM62, 2.00, -0.10, +0.10, noshift, amax=0.0\n",
      "8 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "rescale = 1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "  200 / 203\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP   181 /202 (thread    7) \n",
      "Reallocating (by thread 4) ..done. *alloclen = 2463\n",
      "STEP   202 /202 (thread    2) \n",
      "done.\n",
      "tbfast (aa) Version 7.458\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "8 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 8\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "rescale = 1\n",
      "\n",
      "  200 / 203\n",
      "Segment   1/  1    1-1271\n",
      "003-0402-0 (thread    6) better        \n",
      "Converged2.\n",
      "done\n",
      "dvtditr (aa) Version 7.458\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "8 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "(phylogenetics) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "linsi --thread 8 data/\"$inseq\"_selection-algae_guidev3.fasta > data/alignments_raw/\"$inseq\"_selection-algae_guidev3_aligned-mafft-linsi.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "rescale = 1\n",
      "All-to-all alignment.\n",
      "tbfast-pair (aa) Version 7.458\n",
      "alg=A, model=BLOSUM62, 2.00, -0.10, +0.10, noshift, amax=0.0\n",
      "8 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "rescale = 1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "  200 / 203\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP   180 /202 (thread    3) \n",
      "Reallocating (by thread 2) ..done. *alloclen = 2547\n",
      "STEP   202 /202 (thread    7) \n",
      "done.\n",
      "tbfast (aa) Version 7.458\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "8 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 8\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "rescale = 1\n",
      "\n",
      "  200 / 203\n",
      "Segment   1/  1    1-1266\n",
      "003-0402-0 (thread    1) worse         \n",
      "Converged2.\n",
      "done\n",
      "dvtditr (aa) Version 7.458\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "8 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " G-INS-i (Suitable for sequences of similar lengths, very slow)\n",
      " Iterative refinement method (<16) with GLOBAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "(phylogenetics) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ginsi --thread 8 data/\"$inseq\"_selection-algae_guidev3.fasta > data/alignments_raw/\"$inseq\"_selection-algae_guidev3_aligned-mafft-ginsi.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calling DASH (https://sysimm.org/dash/)\n",
      "------------------\n",
      "MAFFT-DASH Client v 1.1\n",
      "------------------\n",
      "Querying from https://sysimm.org/dash/REST1.0/\n",
      "Building query for template selection.\n",
      "Sending query for template selection...\n",
      "Waiting for response from server...\n",
      "Querying DASH for domains - [0%]\n",
      "Querying DASH for domains - [100%]\n",
      "Querying DASH for sequences - [0%]\n",
      "Querying DASH for sequences - [100%]\n",
      "Downloading alignments - [0%]\n",
      "Downloading alignments - [100%]\n",
      "Combining domain alignments - [0%]\n",
      "Combining domain alignments - [100%]\n",
      "Filtering structural restraint(hat3) file...\n",
      "Combining original sequences with DASH sequences...\n",
      "------------------\n",
      "Ready to run MAFFT:\n",
      "  mafft --seedtable hat3.seed --localpair --maxiterate 100 dashsequences\n",
      "Done.\n",
      "# of structures = \n",
      "216\n",
      "outputhat23=3\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "rescale = 1\n",
      "All-to-all alignment.\n",
      "tbfast-pair (aa) Version 7.458\n",
      "alg=A, model=BLOSUM62, 2.00, -0.10, +0.10, noshift, amax=0.0\n",
      "8 thread(s)\n",
      "\n",
      "outputhat23=3\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "rescale = 1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "  210 / 216\n",
      "   10 / 13\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP   201 /215 (thread    6) \n",
      "Reallocating (by thread 1) ..done. *alloclen = 2560\n",
      "STEP   215 /215 (thread    1) \n",
      "done.\n",
      "tbfast (aa) Version 7.458\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "8 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 8\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 3\n",
      "sueff_global = 0.100000\n",
      "nadd = 3\n",
      "Loading 'hat3' ... done.\n",
      "rescale = 1\n",
      "\n",
      "  210 / 216\n",
      "Segment   1/  1    1-1251\n",
      "003-0428-0 (thread    1) better        \n",
      "Converged2.\n",
      "\n",
      "Reached 3\n",
      "done\n",
      "dvtditr (aa) Version 7.458\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "8 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " G-INS-i (Suitable for sequences of similar lengths, very slow)\n",
      " Iterative refinement method (<3) with GLOBAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "(phylogenetics) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "linsi --dash --thread 8 data/\"$inseq\"_selection-algae_guidev3.fasta > data/alignments_raw/\"$inseq\"_selection-algae_guidev3_aligned-mafft-linsi-dash.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "data/alignments_raw/MIKC_orthogroup_selection-algae_guidev3_aligned-mafft-ginsi.fasta\n",
      "data/alignments_raw/MIKC_orthogroup_selection-algae_guidev3_aligned-mafft-linsi-dash.fasta\n",
      "data/alignments_raw/MIKC_orthogroup_selection-algae_guidev3_aligned-mafft-linsi.fasta\n",
      "(phylogenetics) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls data/alignments_raw/MIKC_orthogroup_selection-algae_guidev3*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(phylogenetics) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "inseq=\"$inseq\"_selection-algae_guidev3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "trimAl v1.4.rev15 build[2013-12-17]. 2009-2013. Salvador Capella-Gutierrez and Toni Gabald√≥n.\n",
      "\n",
      "trimAl webpage: http://trimal.cgenomics.org\n",
      "\n",
      "This program is free software: you can redistribute it and/or modify \n",
      "it under the terms of the GNU General Public License as published by \n",
      "the Free Software Foundation, the last available version.\n",
      "\n",
      "Please cite:\n",
      "\t\ttrimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses.\n",
      "\t\tSalvador Capella-Gutierrez; Jose M. Silla-Martinez; Toni Gabaldon.\n",
      "\t\tBioinformatics 2009, 25:1972-1973.\n",
      "\n",
      "Basic usage\n",
      "\ttrimal -in <inputfile> -out <outputfile> -(other options).\n",
      "\n",
      "Common options (for a complete list please see the User Guide or visit http://trimal.cgenomics.org):\n",
      "\n",
      "    -h                       Print this information and show some examples.\n",
      "    --version                Print the trimAl version.\n",
      "\n",
      "    -in <inputfile>          Input file in several formats (clustal, fasta, NBRF/PIR, nexus, phylip3.2, phylip).\n",
      "\n",
      "    -compareset <inputfile>  Input list of paths for the files containing the alignments to compare.\n",
      "    -forceselect <inputfile> Force selection of the given input file in the files comparison method.\n",
      "\n",
      "    -backtrans <inputfile>   Use a Coding Sequences file to get a backtranslation for a given AA alignment\n",
      "    -ignorestopcodon         Ignore stop codons in the input coding sequences\n",
      "    -splitbystopcodon        Split input coding sequences up to first stop codon appearance\n",
      "\n",
      "    -matrix <inpufile>       Input file for user-defined similarity matrix (default is Blosum62).\n",
      "\n",
      "    -out <outputfile>        Output alignment in the same input format (default stdout). (default input format)\n",
      "    -htmlout <outputfile>    Get a summary of trimal's work in an HTML file.\n",
      "\n",
      "    -keepheader              Keep original sequence header including non-alphanumeric characters.\n",
      "                             Only available for input FASTA format files. (future versions will extend this feature)\n",
      "\n",
      "    -nbrf                    Output file in NBRF/PIR format\n",
      "    -mega                    Output file in MEGA format\n",
      "    -nexus                   Output file in NEXUS format\n",
      "    -clustal                 Output file in CLUSTAL format\n",
      "\n",
      "    -fasta                   Output file in FASTA format\n",
      "    -fasta_m10               Output file in FASTA format. Sequences name length up to 10 characters.\n",
      "\n",
      "    -phylip                  Output file in PHYLIP/PHYLIP4 format\n",
      "    -phylip_m10              Output file in PHYLIP/PHYLIP4 format. Sequences name length up to 10 characters.\n",
      "    -phylip_paml             Output file in PHYLIP format compatible with PAML\n",
      "    -phylip_paml_m10         Output file in PHYLIP format compatible with PAML. Sequences name length up to 10 characters.\n",
      "    -phylip3.2               Output file in PHYLIP3.2 format\n",
      "    -phylip3.2_m10           Output file in PHYLIP3.2 format. Sequences name length up to 10 characters.\n",
      "\n",
      "    -complementary           Get the complementary alignment.\n",
      "    -colnumbering            Get the relationship between the columns in the old and new alignment.\n",
      "\n",
      "    -selectcols { n,l,m-k }  Selection of columns to be removed from the alignment. Range: [0 - (Number of Columns - 1)]. (see User Guide).\n",
      "    -selectseqs { n,l,m-k }  Selection of sequences to be removed from the alignment. Range: [0 - (Number of Sequences - 1)]. (see User Guide).\n",
      "\n",
      "    -gt -gapthreshold <n>    1 - (fraction of sequences with a gap allowed). Range: [0 - 1]\n",
      "    -st -simthreshold <n>    Minimum average similarity allowed. Range: [0 - 1]\n",
      "    -ct -conthreshold <n>    Minimum consistency value allowed.Range: [0 - 1]\n",
      "    -cons <n>                Minimum percentage of the positions in the original alignment to conserve. Range: [0 - 100]\n",
      "\n",
      "    -nogaps                  Remove all positions with gaps in the alignment.\n",
      "    -noallgaps               Remove columns composed only by gaps.\n",
      "    -keepseqs                Keep sequences even if they are composed only by gaps.\n",
      "\n",
      "    -gappyout                Use automated selection on \"gappyout\" mode. This method only uses information based on gaps' distribution. (see User Guide).\n",
      "    -strict                  Use automated selection on \"strict\" mode. (see User Guide).\n",
      "    -strictplus              Use automated selection on \"strictplus\" mode. (see User Guide).\n",
      "                             (Optimized for Neighbour Joining phylogenetic tree reconstruction).\n",
      "\n",
      "    -automated1              Use a heuristic selection of the automatic method based on similarity statistics. (see User Guide). (Optimized for Maximum Likelihood phylogenetic tree reconstruction).\n",
      "\n",
      "    -terminalonly            Only columns out of internal boundaries (first and last column without gaps) are \n",
      "                             candidated to be trimmed depending on the applied method\n",
      "    -block <n>               Minimum column block size to be kept in the trimmed alignment. Available with manual and automatic (gappyout) methods\n",
      "\n",
      "    -resoverlap              Minimum overlap of a positions with other positions in the column to be considered a \"good position\". Range: [0 - 1]. (see User Guide).\n",
      "    -seqoverlap              Minimum percentage of \"good positions\" that a sequence must have in order to be conserved. Range: [0 - 100](see User Guide).\n",
      "\n",
      "    -clusters <n>            Get the most Nth representatives sequences from a given alignment. Range: [1 - (Number of sequences)]\n",
      "    -maxidentity <n>         Get the representatives sequences for a given identity threshold. Range: [0 - 1].\n",
      "\n",
      "    -w <n>                   (half) Window size, score of position i is the average of the window (i - n) to (i + n).\n",
      "    -gw <n>                  (half) Window size only applies to statistics/methods based on Gaps.\n",
      "    -sw <n>                  (half) Window size only applies to statistics/methods based on Similarity.\n",
      "    -cw <n>                  (half) Window size only applies to statistics/methods based on Consistency.\n",
      "\n",
      "    -sgc                     Print gap scores for each column in the input alignment.\n",
      "    -sgt                     Print accumulated gap scores for the input alignment.\n",
      "    -ssc                     Print similarity scores for each column in the input alignment.\n",
      "    -sst                     Print accumulated similarity scores for the input alignment.\n",
      "    -sfc                     Print sum-of-pairs scores for each column from the selected alignment\n",
      "    -sft                     Print accumulated sum-of-pairs scores for the selected alignment\n",
      "    -sident                  Print identity scores for all sequences in the input alignment. (see User Guide).\n",
      "\n",
      "Some Examples:\n",
      "\n",
      "1) Removes all positions in the alignment with gaps in 10% or more of\n",
      "   the sequences, unless this leaves less than 60% of original alignment. \n",
      "   In such case, print the 60% best (with less gaps) positions.\n",
      "\n",
      "   trimal -in <inputfile> -out <outputfile> -gt 0.9 -cons 60\n",
      "\n",
      "2) As above but, the gap score is averaged over a window starting\n",
      "   3 positions before and ending 3 positions after each column.\n",
      "\n",
      "   trimal -in <inputfile> -out <outputfile> -gt 0.9 -cons 60 -w 3\n",
      "\n",
      "3) Use an automatic method to decide optimal thresholds, based in the gap scores\n",
      "   from input alignment. (see User Guide for details).\n",
      "\n",
      "   trimal -in <inputfile> -out <outputfile> -gappyout\n",
      "\n",
      "4) Use automatic methods to decide optimal thresholds, based on the combination \n",
      "   of gap and similarity scores. (see User Guide for details).\n",
      "\n",
      "   trimal -in <inputfile> -out <outputfile> -strictplus\n",
      "\n",
      "5) Use an heuristic to decide the optimal method for trimming the alignment. \n",
      "   (see User Guide for details).\n",
      "\n",
      "   trimal -in <inputfile> -out <outputfile> -automated1\n",
      "\n",
      "6) Use residues and sequences overlap thresholds to delete some sequences from the \n",
      "   alignemnt. (see User Guide for details).\n",
      "\n",
      "   trimal -in <inputfile> -out <outputfile> -resoverlap 0.8 -seqoverlap 75\n",
      "\n",
      "7) Selection of columns to be deleted from the alignment. The selection can \n",
      "   be a column number or a column number interval. Start from 0\n",
      "\n",
      "   trimal -in <inputfile> -out <outputfile> -selectcols { 0,2,3,10,45-60,68,70-78 }\n",
      "\n",
      "8) Get the complementary alignment from the alignment previously trimmed.\n",
      "\n",
      "   trimal -in <inputfile> -out <outputfile> -selectcols { 0,2,3,10,45-60,68,70-78 } -complementary\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "9) Selection of sequences to be deleted from the alignment. Start in 0\n",
      "\n",
      "   trimal -in <inputfile> -out <outputfile> -selectseqs { 2,4,8-12 } \n",
      "\n",
      "10) Select the 5 most representative sequences from the alignment\n",
      "\n",
      "   trimal -in <inputfile> -out <outputfile> -clusters 5 \n",
      "\n",
      "(phylogenetics) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "trimal -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(phylogenetics) (phylogenetics) (phylogenetics) (phylogenetics) (phylogenetics) (phylogenetics) trimming alignment data/alignments_raw/MIKC_orthogroup_selection-algae_guidev3_aligned-mafft-ginsi.fasta\n",
      "\n",
      "WARNING: Removing column '440' composed only by gaps\n",
      "WARNING: Removing column '441' composed only by gaps\n",
      "WARNING: Removing column '442' composed only by gaps\n",
      "WARNING: Removing column '443' composed only by gaps\n",
      "WARNING: Removing column '444' composed only by gaps\n",
      "WARNING: Removing column '445' composed only by gaps\n",
      "WARNING: Removing column '446' composed only by gaps\n",
      "WARNING: Removing column '447' composed only by gaps\n",
      "WARNING: Removing column '448' composed only by gaps\n",
      "WARNING: Removing column '596' composed only by gaps\n",
      "WARNING: Removing column '957' composed only by gaps\n",
      "WARNING: Removing column '973' composed only by gaps\n",
      "WARNING: Removing column '974' composed only by gaps\n",
      "WARNING: Removing column '975' composed only by gaps\n",
      "WARNING: Removing column '976' composed only by gaps\n",
      "WARNING: Removing column '977' composed only by gaps\n",
      "WARNING: Removing column '978' composed only by gaps\n",
      "WARNING: Removing column '979' composed only by gaps\n",
      "WARNING: Removing column '980' composed only by gaps\n",
      "WARNING: Removing column '983' composed only by gaps\n",
      "WARNING: Removing column '985' composed only by gaps\n",
      "WARNING: Removing column '986' composed only by gaps\n",
      "WARNING: Removing column '987' composed only by gaps\n",
      "WARNING: Removing column '988' composed only by gaps\n",
      "WARNING: Removing column '989' composed only by gaps\n",
      "WARNING: Removing column '990' composed only by gaps\n",
      "WARNING: Removing column '991' composed only by gaps\n",
      "WARNING: Removing column '992' composed only by gaps\n",
      "WARNING: Removing column '993' composed only by gaps\n",
      "WARNING: Removing column '994' composed only by gaps\n",
      "WARNING: Removing column '995' composed only by gaps\n",
      "WARNING: Removing column '996' composed only by gaps\n",
      "WARNING: Removing column '997' composed only by gaps\n",
      "WARNING: Removing column '998' composed only by gaps\n",
      "WARNING: Removing column '999' composed only by gaps\n",
      "WARNING: Removing column '1000' composed only by gaps\n",
      "WARNING: Removing column '1009' composed only by gaps\n",
      "WARNING: Removing column '1010' composed only by gaps\n",
      "WARNING: Removing column '1011' composed only by gaps\n",
      "WARNING: Removing column '1012' composed only by gaps\n",
      "WARNING: Removing column '1013' composed only by gaps\n",
      "WARNING: Removing column '1014' composed only by gaps\n",
      "WARNING: Removing column '1015' composed only by gaps\n",
      "WARNING: Removing column '1016' composed only by gaps\n",
      "WARNING: Removing column '1017' composed only by gaps\n",
      "WARNING: Removing column '1018' composed only by gaps\n",
      "WARNING: Removing column '1019' composed only by gaps\n",
      "WARNING: Removing column '1020' composed only by gaps\n",
      "WARNING: Removing column '1058' composed only by gaps\n",
      "WARNING: Removing column '1059' composed only by gaps\n",
      "WARNING: Removing column '1060' composed only by gaps\n",
      "WARNING: Removing column '1065' composed only by gaps\n",
      "WARNING: Removing column '1066' composed only by gaps\n",
      "WARNING: Removing column '1067' composed only by gaps\n",
      "WARNING: Removing column '1068' composed only by gaps\n",
      "WARNING: Removing column '1069' composed only by gaps\n",
      "WARNING: Removing column '1070' composed only by gaps\n",
      "WARNING: Removing column '1071' composed only by gaps\n",
      "WARNING: Removing column '1072' composed only by gaps\n",
      "WARNING: Removing column '1073' composed only by gaps\n",
      "WARNING: Removing column '1074' composed only by gaps\n",
      "WARNING: Removing column '1075' composed only by gaps\n",
      "WARNING: Removing column '1076' composed only by gaps\n",
      "WARNING: Removing column '1077' composed only by gaps\n",
      "WARNING: Removing column '1078' composed only by gaps\n",
      "WARNING: Removing column '1079' composed only by gaps\n",
      "WARNING: Removing column '1082' composed only by gaps\n",
      "WARNING: Removing column '1083' composed only by gaps\n",
      "WARNING: Removing column '1084' composed only by gaps\n",
      "WARNING: Removing column '1085' composed only by gaps\n",
      "WARNING: Removing column '1086' composed only by gaps\n",
      "WARNING: Removing column '1087' composed only by gaps\n",
      "WARNING: Removing column '1088' composed only by gaps\n",
      "WARNING: Removing column '1089' composed only by gaps\n",
      "WARNING: Removing column '1090' composed only by gaps\n",
      "WARNING: Removing column '1091' composed only by gaps\n",
      "WARNING: Removing column '1092' composed only by gaps\n",
      "WARNING: Removing column '1093' composed only by gaps\n",
      "WARNING: Removing column '1094' composed only by gaps\n",
      "WARNING: Removing column '1095' composed only by gaps\n",
      "WARNING: Removing column '1096' composed only by gaps\n",
      "WARNING: Removing column '1097' composed only by gaps\n",
      "WARNING: Removing column '1098' composed only by gaps\n",
      "WARNING: Removing column '1099' composed only by gaps\n",
      "WARNING: Removing column '1100' composed only by gaps\n",
      "WARNING: Removing column '1101' composed only by gaps\n",
      "WARNING: Removing column '1102' composed only by gaps\n",
      "WARNING: Removing column '1103' composed only by gaps\n",
      "WARNING: Removing column '1104' composed only by gaps\n",
      "WARNING: Removing column '1105' composed only by gaps\n",
      "WARNING: Removing column '1106' composed only by gaps\n",
      "WARNING: Removing column '1107' composed only by gaps\n",
      "WARNING: Removing column '1108' composed only by gaps\n",
      "WARNING: Removing column '1109' composed only by gaps\n",
      "WARNING: Removing column '1110' composed only by gaps\n",
      "WARNING: Removing column '1111' composed only by gaps\n",
      "WARNING: Removing column '1112' composed only by gaps\n",
      "WARNING: Removing column '1113' composed only by gaps\n",
      "WARNING: Removing column '1114' composed only by gaps\n",
      "WARNING: Removing column '1115' composed only by gaps\n",
      "WARNING: Removing column '1116' composed only by gaps\n",
      "WARNING: Removing column '1117' composed only by gaps\n",
      "WARNING: Removing column '1118' composed only by gaps\n",
      "WARNING: Removing column '1119' composed only by gaps\n",
      "WARNING: Removing column '1120' composed only by gaps\n",
      "WARNING: Removing column '1121' composed only by gaps\n",
      "WARNING: Removing column '1122' composed only by gaps\n",
      "WARNING: Removing column '1123' composed only by gaps\n",
      "WARNING: Removing column '1124' composed only by gaps\n",
      "WARNING: Removing column '1146' composed only by gaps\n",
      "WARNING: Removing column '1147' composed only by gaps\n",
      "WARNING: Removing column '1148' composed only by gaps\n",
      "WARNING: Removing column '1149' composed only by gaps\n",
      "WARNING: Removing column '1150' composed only by gaps\n",
      "WARNING: Removing column '1151' composed only by gaps\n",
      "WARNING: Removing column '1152' composed only by gaps\n",
      "WARNING: Removing column '1153' composed only by gaps\n",
      "WARNING: Removing column '1154' composed only by gaps\n",
      "WARNING: Removing column '1155' composed only by gaps\n",
      "WARNING: Removing column '1156' composed only by gaps\n",
      "WARNING: Removing column '1157' composed only by gaps\n",
      "WARNING: Removing column '1158' composed only by gaps\n",
      "WARNING: Removing column '1159' composed only by gaps\n",
      "WARNING: Removing column '1160' composed only by gaps\n",
      "WARNING: Removing column '1161' composed only by gaps\n",
      "WARNING: Removing column '1162' composed only by gaps\n",
      "WARNING: Removing column '1163' composed only by gaps\n",
      "WARNING: Removing column '1164' composed only by gaps\n",
      "WARNING: Removing column '1165' composed only by gaps\n",
      "WARNING: Removing column '1166' composed only by gaps\n",
      "WARNING: Removing column '1167' composed only by gaps\n",
      "WARNING: Removing column '1168' composed only by gaps\n",
      "WARNING: Removing column '1169' composed only by gaps\n",
      "WARNING: Removing column '1170' composed only by gaps\n",
      "WARNING: Removing column '1171' composed only by gaps\n",
      "WARNING: Removing column '1172' composed only by gaps\n",
      "WARNING: Removing column '1173' composed only by gaps\n",
      "WARNING: Removing column '1174' composed only by gaps\n",
      "WARNING: Removing column '1175' composed only by gaps\n",
      "WARNING: Removing column '1176' composed only by gaps\n",
      "WARNING: Removing column '1177' composed only by gaps\n",
      "WARNING: Removing column '1178' composed only by gaps\n",
      "WARNING: Removing column '1179' composed only by gaps\n",
      "WARNING: Removing column '1180' composed only by gaps\n",
      "WARNING: Removing column '1181' composed only by gaps\n",
      "WARNING: Removing column '1182' composed only by gaps\n",
      "WARNING: Removing column '1183' composed only by gaps\n",
      "WARNING: Removing column '1184' composed only by gaps\n",
      "WARNING: Removing column '1185' composed only by gaps\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Removing column '1212' composed only by gaps\n",
      "WARNING: Removing column '1213' composed only by gaps\n",
      "WARNING: Removing column '1214' composed only by gaps\n",
      "WARNING: Removing column '1215' composed only by gaps\n",
      "WARNING: Removing column '1216' composed only by gaps\n",
      "WARNING: Removing column '1217' composed only by gaps\n",
      "WARNING: Removing column '1218' composed only by gaps\n",
      "WARNING: Removing column '1219' composed only by gaps\n",
      "WARNING: Removing column '1220' composed only by gaps\n",
      "WARNING: Removing column '1221' composed only by gaps\n",
      "WARNING: Removing column '1222' composed only by gaps\n",
      "\n",
      "trimming alignment data/alignments_raw/MIKC_orthogroup_selection-algae_guidev3_aligned-mafft-linsi-dash.fasta\n",
      "\n",
      "WARNING: Removing column '437' composed only by gaps\n",
      "WARNING: Removing column '438' composed only by gaps\n",
      "WARNING: Removing column '463' composed only by gaps\n",
      "WARNING: Removing column '464' composed only by gaps\n",
      "WARNING: Removing column '465' composed only by gaps\n",
      "WARNING: Removing column '466' composed only by gaps\n",
      "WARNING: Removing column '467' composed only by gaps\n",
      "WARNING: Removing column '468' composed only by gaps\n",
      "WARNING: Removing column '469' composed only by gaps\n",
      "WARNING: Removing column '470' composed only by gaps\n",
      "WARNING: Removing column '471' composed only by gaps\n",
      "WARNING: Removing column '472' composed only by gaps\n",
      "WARNING: Removing column '473' composed only by gaps\n",
      "WARNING: Removing column '510' composed only by gaps\n",
      "WARNING: Removing column '511' composed only by gaps\n",
      "WARNING: Removing column '512' composed only by gaps\n",
      "WARNING: Removing column '513' composed only by gaps\n",
      "WARNING: Removing column '670' composed only by gaps\n",
      "WARNING: Removing column '671' composed only by gaps\n",
      "WARNING: Removing column '672' composed only by gaps\n",
      "WARNING: Removing column '673' composed only by gaps\n",
      "WARNING: Removing column '674' composed only by gaps\n",
      "WARNING: Removing column '675' composed only by gaps\n",
      "WARNING: Removing column '679' composed only by gaps\n",
      "WARNING: Removing column '680' composed only by gaps\n",
      "WARNING: Removing column '681' composed only by gaps\n",
      "WARNING: Removing column '682' composed only by gaps\n",
      "WARNING: Removing column '683' composed only by gaps\n",
      "WARNING: Removing column '691' composed only by gaps\n",
      "WARNING: Removing column '792' composed only by gaps\n",
      "WARNING: Removing column '793' composed only by gaps\n",
      "WARNING: Removing column '825' composed only by gaps\n",
      "WARNING: Removing column '826' composed only by gaps\n",
      "WARNING: Removing column '827' composed only by gaps\n",
      "WARNING: Removing column '828' composed only by gaps\n",
      "WARNING: Removing column '829' composed only by gaps\n",
      "WARNING: Removing column '836' composed only by gaps\n",
      "WARNING: Removing column '867' composed only by gaps\n",
      "WARNING: Removing column '871' composed only by gaps\n",
      "WARNING: Removing column '872' composed only by gaps\n",
      "WARNING: Removing column '873' composed only by gaps\n",
      "WARNING: Removing column '874' composed only by gaps\n",
      "WARNING: Removing column '875' composed only by gaps\n",
      "WARNING: Removing column '876' composed only by gaps\n",
      "WARNING: Removing column '877' composed only by gaps\n",
      "WARNING: Removing column '878' composed only by gaps\n",
      "WARNING: Removing column '879' composed only by gaps\n",
      "WARNING: Removing column '880' composed only by gaps\n",
      "WARNING: Removing column '881' composed only by gaps\n",
      "WARNING: Removing column '882' composed only by gaps\n",
      "WARNING: Removing column '883' composed only by gaps\n",
      "WARNING: Removing column '884' composed only by gaps\n",
      "WARNING: Removing column '885' composed only by gaps\n",
      "WARNING: Removing column '886' composed only by gaps\n",
      "WARNING: Removing column '887' composed only by gaps\n",
      "WARNING: Removing column '888' composed only by gaps\n",
      "WARNING: Removing column '892' composed only by gaps\n",
      "WARNING: Removing column '893' composed only by gaps\n",
      "WARNING: Removing column '894' composed only by gaps\n",
      "WARNING: Removing column '895' composed only by gaps\n",
      "WARNING: Removing column '896' composed only by gaps\n",
      "WARNING: Removing column '897' composed only by gaps\n",
      "WARNING: Removing column '898' composed only by gaps\n",
      "WARNING: Removing column '899' composed only by gaps\n",
      "WARNING: Removing column '902' composed only by gaps\n",
      "WARNING: Removing column '903' composed only by gaps\n",
      "WARNING: Removing column '904' composed only by gaps\n",
      "WARNING: Removing column '905' composed only by gaps\n",
      "WARNING: Removing column '906' composed only by gaps\n",
      "WARNING: Removing column '907' composed only by gaps\n",
      "WARNING: Removing column '908' composed only by gaps\n",
      "WARNING: Removing column '909' composed only by gaps\n",
      "WARNING: Removing column '910' composed only by gaps\n",
      "WARNING: Removing column '913' composed only by gaps\n",
      "WARNING: Removing column '914' composed only by gaps\n",
      "WARNING: Removing column '915' composed only by gaps\n",
      "WARNING: Removing column '916' composed only by gaps\n",
      "WARNING: Removing column '917' composed only by gaps\n",
      "WARNING: Removing column '918' composed only by gaps\n",
      "WARNING: Removing column '919' composed only by gaps\n",
      "WARNING: Removing column '920' composed only by gaps\n",
      "WARNING: Removing column '921' composed only by gaps\n",
      "WARNING: Removing column '922' composed only by gaps\n",
      "WARNING: Removing column '923' composed only by gaps\n",
      "WARNING: Removing column '924' composed only by gaps\n",
      "WARNING: Removing column '925' composed only by gaps\n",
      "WARNING: Removing column '926' composed only by gaps\n",
      "WARNING: Removing column '927' composed only by gaps\n",
      "WARNING: Removing column '928' composed only by gaps\n",
      "WARNING: Removing column '929' composed only by gaps\n",
      "WARNING: Removing column '930' composed only by gaps\n",
      "WARNING: Removing column '931' composed only by gaps\n",
      "WARNING: Removing column '932' composed only by gaps\n",
      "WARNING: Removing column '933' composed only by gaps\n",
      "WARNING: Removing column '978' composed only by gaps\n",
      "WARNING: Removing column '979' composed only by gaps\n",
      "WARNING: Removing column '980' composed only by gaps\n",
      "WARNING: Removing column '981' composed only by gaps\n",
      "WARNING: Removing column '982' composed only by gaps\n",
      "WARNING: Removing column '983' composed only by gaps\n",
      "WARNING: Removing column '984' composed only by gaps\n",
      "WARNING: Removing column '985' composed only by gaps\n",
      "WARNING: Removing column '986' composed only by gaps\n",
      "WARNING: Removing column '1061' composed only by gaps\n",
      "WARNING: Removing column '1062' composed only by gaps\n",
      "WARNING: Removing column '1063' composed only by gaps\n",
      "WARNING: Removing column '1064' composed only by gaps\n",
      "WARNING: Removing column '1065' composed only by gaps\n",
      "WARNING: Removing column '1072' composed only by gaps\n",
      "WARNING: Removing column '1073' composed only by gaps\n",
      "WARNING: Removing column '1074' composed only by gaps\n",
      "WARNING: Removing column '1075' composed only by gaps\n",
      "WARNING: Removing column '1076' composed only by gaps\n",
      "WARNING: Removing column '1077' composed only by gaps\n",
      "WARNING: Removing column '1078' composed only by gaps\n",
      "WARNING: Removing column '1079' composed only by gaps\n",
      "WARNING: Removing column '1080' composed only by gaps\n",
      "WARNING: Removing column '1081' composed only by gaps\n",
      "WARNING: Removing column '1082' composed only by gaps\n",
      "WARNING: Removing column '1083' composed only by gaps\n",
      "WARNING: Removing column '1084' composed only by gaps\n",
      "WARNING: Removing column '1085' composed only by gaps\n",
      "WARNING: Removing column '1086' composed only by gaps\n",
      "WARNING: Removing column '1087' composed only by gaps\n",
      "WARNING: Removing column '1088' composed only by gaps\n",
      "WARNING: Removing column '1089' composed only by gaps\n",
      "WARNING: Removing column '1090' composed only by gaps\n",
      "WARNING: Removing column '1091' composed only by gaps\n",
      "WARNING: Removing column '1092' composed only by gaps\n",
      "WARNING: Removing column '1093' composed only by gaps\n",
      "WARNING: Removing column '1094' composed only by gaps\n",
      "WARNING: Removing column '1095' composed only by gaps\n",
      "WARNING: Removing column '1096' composed only by gaps\n",
      "WARNING: Removing column '1115' composed only by gaps\n",
      "WARNING: Removing column '1116' composed only by gaps\n",
      "WARNING: Removing column '1117' composed only by gaps\n",
      "WARNING: Removing column '1118' composed only by gaps\n",
      "\n",
      "trimming alignment data/alignments_raw/MIKC_orthogroup_selection-algae_guidev3_aligned-mafft-linsi.fasta\n",
      "\n",
      "WARNING: Removing column '478' composed only by gaps\n",
      "WARNING: Removing column '479' composed only by gaps\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Removing column '480' composed only by gaps\n",
      "WARNING: Removing column '481' composed only by gaps\n",
      "WARNING: Removing column '482' composed only by gaps\n",
      "WARNING: Removing column '483' composed only by gaps\n",
      "WARNING: Removing column '484' composed only by gaps\n",
      "WARNING: Removing column '485' composed only by gaps\n",
      "WARNING: Removing column '486' composed only by gaps\n",
      "WARNING: Removing column '506' composed only by gaps\n",
      "WARNING: Removing column '507' composed only by gaps\n",
      "WARNING: Removing column '508' composed only by gaps\n",
      "WARNING: Removing column '509' composed only by gaps\n",
      "WARNING: Removing column '510' composed only by gaps\n",
      "WARNING: Removing column '511' composed only by gaps\n",
      "WARNING: Removing column '512' composed only by gaps\n",
      "WARNING: Removing column '513' composed only by gaps\n",
      "WARNING: Removing column '514' composed only by gaps\n",
      "WARNING: Removing column '515' composed only by gaps\n",
      "WARNING: Removing column '516' composed only by gaps\n",
      "WARNING: Removing column '553' composed only by gaps\n",
      "WARNING: Removing column '554' composed only by gaps\n",
      "WARNING: Removing column '555' composed only by gaps\n",
      "WARNING: Removing column '556' composed only by gaps\n",
      "WARNING: Removing column '725' composed only by gaps\n",
      "WARNING: Removing column '726' composed only by gaps\n",
      "WARNING: Removing column '727' composed only by gaps\n",
      "WARNING: Removing column '728' composed only by gaps\n",
      "WARNING: Removing column '729' composed only by gaps\n",
      "WARNING: Removing column '730' composed only by gaps\n",
      "WARNING: Removing column '731' composed only by gaps\n",
      "WARNING: Removing column '732' composed only by gaps\n",
      "WARNING: Removing column '733' composed only by gaps\n",
      "WARNING: Removing column '734' composed only by gaps\n",
      "WARNING: Removing column '735' composed only by gaps\n",
      "WARNING: Removing column '736' composed only by gaps\n",
      "WARNING: Removing column '737' composed only by gaps\n",
      "WARNING: Removing column '741' composed only by gaps\n",
      "WARNING: Removing column '749' composed only by gaps\n",
      "WARNING: Removing column '781' composed only by gaps\n",
      "WARNING: Removing column '782' composed only by gaps\n",
      "WARNING: Removing column '783' composed only by gaps\n",
      "WARNING: Removing column '784' composed only by gaps\n",
      "WARNING: Removing column '785' composed only by gaps\n",
      "WARNING: Removing column '786' composed only by gaps\n",
      "WARNING: Removing column '787' composed only by gaps\n",
      "WARNING: Removing column '930' composed only by gaps\n",
      "WARNING: Removing column '931' composed only by gaps\n",
      "WARNING: Removing column '932' composed only by gaps\n",
      "WARNING: Removing column '933' composed only by gaps\n",
      "WARNING: Removing column '934' composed only by gaps\n",
      "WARNING: Removing column '935' composed only by gaps\n",
      "WARNING: Removing column '936' composed only by gaps\n",
      "WARNING: Removing column '937' composed only by gaps\n",
      "WARNING: Removing column '938' composed only by gaps\n",
      "WARNING: Removing column '943' composed only by gaps\n",
      "WARNING: Removing column '944' composed only by gaps\n",
      "WARNING: Removing column '945' composed only by gaps\n",
      "WARNING: Removing column '946' composed only by gaps\n",
      "WARNING: Removing column '963' composed only by gaps\n",
      "WARNING: Removing column '964' composed only by gaps\n",
      "WARNING: Removing column '965' composed only by gaps\n",
      "WARNING: Removing column '966' composed only by gaps\n",
      "WARNING: Removing column '967' composed only by gaps\n",
      "WARNING: Removing column '968' composed only by gaps\n",
      "WARNING: Removing column '969' composed only by gaps\n",
      "WARNING: Removing column '970' composed only by gaps\n",
      "WARNING: Removing column '971' composed only by gaps\n",
      "WARNING: Removing column '972' composed only by gaps\n",
      "WARNING: Removing column '973' composed only by gaps\n",
      "WARNING: Removing column '974' composed only by gaps\n",
      "WARNING: Removing column '975' composed only by gaps\n",
      "WARNING: Removing column '976' composed only by gaps\n",
      "WARNING: Removing column '977' composed only by gaps\n",
      "WARNING: Removing column '978' composed only by gaps\n",
      "WARNING: Removing column '979' composed only by gaps\n",
      "WARNING: Removing column '980' composed only by gaps\n",
      "WARNING: Removing column '981' composed only by gaps\n",
      "WARNING: Removing column '982' composed only by gaps\n",
      "WARNING: Removing column '983' composed only by gaps\n",
      "WARNING: Removing column '984' composed only by gaps\n",
      "WARNING: Removing column '985' composed only by gaps\n",
      "WARNING: Removing column '986' composed only by gaps\n",
      "WARNING: Removing column '987' composed only by gaps\n",
      "WARNING: Removing column '988' composed only by gaps\n",
      "WARNING: Removing column '989' composed only by gaps\n",
      "WARNING: Removing column '990' composed only by gaps\n",
      "WARNING: Removing column '991' composed only by gaps\n",
      "WARNING: Removing column '1131' composed only by gaps\n",
      "WARNING: Removing column '1132' composed only by gaps\n",
      "WARNING: Removing column '1133' composed only by gaps\n",
      "WARNING: Removing column '1134' composed only by gaps\n",
      "WARNING: Removing column '1135' composed only by gaps\n",
      "WARNING: Removing column '1136' composed only by gaps\n",
      "WARNING: Removing column '1159' composed only by gaps\n",
      "WARNING: Removing column '1185' composed only by gaps\n",
      "WARNING: Removing column '1186' composed only by gaps\n",
      "WARNING: Removing column '1187' composed only by gaps\n",
      "WARNING: Removing column '1188' composed only by gaps\n",
      "WARNING: Removing column '1189' composed only by gaps\n",
      "WARNING: Removing column '1190' composed only by gaps\n",
      "WARNING: Removing column '1191' composed only by gaps\n",
      "WARNING: Removing column '1200' composed only by gaps\n",
      "WARNING: Removing column '1201' composed only by gaps\n",
      "WARNING: Removing column '1202' composed only by gaps\n",
      "WARNING: Removing column '1203' composed only by gaps\n",
      "WARNING: Removing column '1204' composed only by gaps\n",
      "\n",
      "(phylogenetics) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "if    [ ! -d data/alignments_trimmed ]\n",
    "then  mkdir  data/alignments_trimmed \n",
    "fi\n",
    "\n",
    "# define appendix only once here:\n",
    "trimappendix='trim-gt1-res9-seq6'\n",
    "\n",
    "\n",
    "for a in \"data/alignments_raw/$inseq\"_aligned*.fasta\n",
    "do  appendix=$(echo $a | cut -d '/' -f 3- | sed \"s/$inseq\\_//\" | sed \"s/.fasta//\")\n",
    "    if    [ ! -f data/alignments_trimmed/\"$inseq\"_\"$appendix\"_\"$trimappendix\".fasta ]\n",
    "    then  echo \"trimming alignment $a\"\n",
    "          sed -i 's/ /_/g' $a\n",
    "          trimal -in $a   \\\n",
    "                 -out data/alignments_trimmed/\"$inseq\"_\"$appendix\"_\"$trimappendix\".fasta \\\n",
    "                 -gt .1 \\\n",
    "                 -resoverlap .9 \\\n",
    "                 -seqoverlap 60 \\\n",
    "                 -htmlout data/alignments_trimmed/\"$inseq\"_\"$appendix\"_\"$trimappendix\".html\n",
    "    fi\n",
    "done"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
